### PR TITLE
add_vagrant_box.sh: move script to contrib/

### PR DESCRIPTION
**Pull request by qmonnet**
We have a script to facilitate the download of the Vagrant boxes. It is currently located under `test/packet/`, because the Packet scripts rely on it as well. Nevertheless, it would be best to move it to a location where it would better indicate that this is a helper script, not something used in the CI (not to be confused with `add_vagrant_box`, with no `.sh` extension).

Let's move the script to `contrib/scripts`. This will place it under the watch of the `cilium/contributing` team.

(The usage blurb from the script is also updated in the documentation.)

I have _not_ tested the CI run on a separate Packet.net instance to validate the change in `test/packet/main.tf` yet.

### CODE REVIEW

File: test/packet/main.tf [hunk 1]
```diff
@@ -46,6 +46,11 @@ resource "packet_device" "test" {
             destination="/provision"
     }
 
+    provisioner "file" {
+            source="../../contrib/scripts/add_vagrant_box.sh"
+            destination="/provision"
+    }
+
```
Comment (pchaigno) [inline test/packet/main.tf:61]: @nebril Is this script still used? If not, should we remove it?

**Review by pchaigno — APPROVED** [review]

File: test/packet/main.tf (hunk 1 above)
Comment (pchaigno) [reply test/packet/main.tf:61]: @qmonnet Not used anymore in testing infra, but André and Chris use it. So we can't remove.

Comment (qmonnet) [reply test/packet/main.tf:61]: Understood, thanks for the follow-up. I'll come back to this PR soon.

**Comment by qmonnet** [comment]
Discussed during 2022-08-03's Community meeting. Let's get this merged, hopefully everything should keep working for Packet if anyone is still using it, otherwise people will yell and we'll fix.

No CI run required here.

**Review by joestringer — APPROVED** [review]

**Pull request merged**